.circleci/config.yml

```
@@ -54,22 +54,26 @@ jobs:
             if [ ! -z "$(alembic branches)" ]; then echo "Multiple alembic heads found"; exit 1; fi
       - run:
           name: Check imports are well organized with isort
+          when: always
           command: venv/bin/isort . --check-only
       - run:
           name: Check code is well formatted with black
+          when: always
           command: venv/bin/black . --check
```

```suggestion
          command: venv/bin/black .
```
En cas de fichiers non-conformes, il me semble que l'on veut justement afficher le détail et ne pas utiliser l'option `--check` cf https://github.com/psf/black#command-line-options

---

Je trouve que ça n'est pas très utile, mais je peux l'ajouter. :) (Par contre la commande sera plutôt `venv/bin/black . --check --diff` pour afficher le diff, car ta commande reformate.)

Je note que l'affichage du diff (même vide !) n'est pas gratuit (17 secondes vs. 0.7s pour un check simple sans diff). En local :

```
$ time black . --check --diff
All done! ✨ 🍰 ✨
1198 files would be left unchanged.
black . --check --diff  55,21s user 1,55s system 332% cpu 17,062 total
$ time black . --check       
All done! ✨ 🍰 ✨
1198 files would be left unchanged.
black . --check  0,59s user 0,07s system 99% cpu 0,662 total

```

---

Mmh, si tu fais ça, ça va surtout corriger les fichiers :)

---

Par contre --diff semble être la bonne option.

Maintenant, ça risque d'output beaucoup de diffs si black n'est pas du tout passé :grimacing:

---

_Bonnes réponses, je testais vos connaissances sur la cli de black._

On pourrait rajouter une étape dans le `nightly`? Par exemple, si `black . --check` retourne `1`, on lance `black . --check --diff`.

---

J'ai mergé sans modifier les arguments de la commande black. Pour ça, cf. la pull request #1574, mais je ne sais pas si ça vaut le coût.